src/internal/hooks/use-mutation-observer/index.ts

```diff
@@ -1,32 +1,18 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import React, { useEffect } from 'react';
-
-import { createSingletonHandler, useStableCallback } from '@cloudscape-design/component-toolkit/internal';
-
-const useMutationSingleton = createSingletonHandler<void>(handler => {
-  const observer = new MutationObserver(() => handler());
-  observer.observe(document.body, { attributes: true, subtree: true });
```

Comment:
this wasn't being actively used anywhere as far as I could tell, and I couldn't understand why it observes on `document.body` (and then does no subtree filtering to actually limit to `elementRef` in the main hook below

Reply:
This is likely a leftover and the current use cases in components if any probably do not require such implementation. However, the same approach is still used in the toolkit (https://github.com/cloudscape-design/component-toolkit/blob/0dd16b944b60f25ec39b4354513fff3009122f6e/src/internal/visual-mode/index.ts#L54) and it is actually needed because we use it in the tools like useCurrentMode that ensure a class-name to be set. As we don't require the class-name to target a specific component, the update fires on any attribute change of any node to inform the requester which then tries to find a parent with the required class name. It is true that the approach is far from being efficient.